### PR TITLE
OLMIS-3448: Alphabetize facilities within dropdowns

### PR DESCRIPTION
**Pull request by jkondrat**

### CODE REVIEW

File: src/referencedata-facility/facility.service.js [hunk 1]
```diff
@@ -307,7 +307,16 @@
              * @return {Promise} Array of facilities with minimal representation
              */
             function getAllMinimal() {
-                return resource.getAllMinimal().$promise;
+                var deferred = $q.defer();
+
+                resource.getAllMinimal(function(facilities) {
+                    var sortedFacilities = _.sortBy(facilities, 'name');
```
Comment (ngraczewski) [inline src/referencedata-facility/facility.service.js]: I don't think this is the responsibility of the facilityService. It probably should be done on the server side.

File: src/referencedata-facility/facility.service.spec.js [hunk 2]
```diff
@@ -183,6 +183,24 @@ describe('facilityService', function() {
             expect(data[0].id).toBe(facilityOne.id);
             expect(data[1].id).toBe(facilityTwo.id);
         });
+
+        it('should sort facilities by name', function() {
+            var data;
+            facilityOne.name = 'Baba';
+            facilityTwo.name = 'Abab';
+
+            $httpBackend.when('GET', referencedataUrlFactory('/api/facilities/minimal')).respond(200, [facilityOne, facilityTwo]);
```
Comment (ngraczewski) [inline src/referencedata-facility/facility.service.spec.js]: Use whenGET instead and move both it and respond to new lines.

File: src/referencedata-facility/facility.service.spec.js [hunk 3]
```diff
@@ -183,6 +183,24 @@ describe('facilityService', function() {
             expect(data[0].id).toBe(facilityOne.id);
             expect(data[1].id).toBe(facilityTwo.id);
         });
+
+        it('should sort facilities by name', function() {
+            var data;
+            facilityOne.name = 'Baba';
```
Comment (ngraczewski) [inline src/referencedata-facility/facility.service.spec.js]: I believe we have fluent builder for facility and we should use them here.

Comment (ngraczewski) [inline src/referencedata-facility/facility.service.spec.js]: I believe we have fluent builder for facility and we should use them here.

File: src/referencedata-facility/facility.service.spec.js (hunk 2 above)
Comment (ngraczewski) [inline src/referencedata-facility/facility.service.spec.js]: Use whenGET instead and move both it and respond to new lines.

File: src/referencedata-facility/facility.service.js (hunk 1 above)
Comment (ngraczewski) [inline src/referencedata-facility/facility.service.js]: I don't think this is the responsibility of the facilityService. It probably should be done on the server side.

**Comment by ngraczewski** [comment]
@mkwiatkowskisoldevelo I agree this should not be done on the UI side.

**Comment by jkondrat** [comment]
@mkwiatkowskisoldevelo @ngraczewski Done

File: src/referencedata-facility/facility.service.js [hunk 4]
```diff
@@ -304,10 +303,17 @@
              * @description
              * Retrieves all facilities with id and name fields.
              *
+             * @param  {Object}  paginationParams the pagination params: page, size, sort
              * @return {Promise} Array of facilities with minimal representation
              */
-            function getAllMinimal() {
-                return resource.getAllMinimal().$promise;
+            function getAllMinimal(paginationParams) {
+                var params = (paginationParams !== undefined) ? paginationParams : {
```
Comment (mkwiatkowskisoldevelo) [inline src/referencedata-facility/facility.service.js]: you don't need to check if it is undefined here, something like if (paginationParams) { would be enough, but I would rather check if sort parameter is set in paginationParamters

File: src/referencedata-facility/facility.service.js [hunk 5]
```diff
@@ -304,10 +303,17 @@
...
+                    'sort': 'name'
```
Comment (mkwiatkowskisoldevelo) [inline src/referencedata-facility/facility.service.js]: please remove quotes around sort

**Pull request merged**